mbuild/box.py

```diff
@@ -61,35 +61,53 @@ def angles(self):
 
     @mins.setter
     def mins(self, mins):
-        if isinstance(mins, list):
-            mins = np.array(mins, dtype=np.float)
+        mins = np.array(mins, dtype=np.float)
         assert mins.shape == (3, )
-        self._mins = mins
-        self._lengths = self.maxs - self.mins
+        self._mins = boxArray(array=mins, var="mins", box=self)
+        self._lengths = boxArray(array=(self.maxs - self.mins), var="lengths", box=self)
 
     @maxs.setter
-    def maxs(self, maxes):
-        if isinstance(maxes, list):
-            maxes = np.array(maxes, dtype=np.float)
-        assert maxes.shape == (3, )
-        self._maxs = maxes
-        self._lengths = self.maxs - self.mins
+    def maxs(self, maxs):
+        maxs = np.array(maxs, dtype=np.float)
+        assert maxs.shape == (3, )
+        self._maxs = boxArray(array=maxs, var="maxs", box=self)
+        self._lengths = boxArray(array= (self.maxs - self.mins), var="lengths", box=self)
 
     @lengths.setter
     def lengths(self, lengths):
-        if isinstance(lengths, list):
-            lengths = np.array(lengths, dtype=np.float)
+        lengths = np.array(lengths, dtype=np.float)
         assert lengths.shape == (3, )
-        self._maxs += 0.5*lengths - 0.5*self.lengths
-        self._mins -= 0.5*lengths - 0.5*self.lengths
-        self._lengths = lengths
+        self._maxs = boxArray(array=(self.maxs + (0.5*lengths - 0.5*self.lengths)), var="maxs", box=self)
+        self._mins = boxArray(array=(self.mins - (0.5*lengths - 0.5*self.lengths)), var="mins", box=self)
+        self._lengths = boxArray(array=lengths, var="lengths", box=self, dtype=np.float)
 
     @angles.setter
     def angles(self, angles):
-        if isinstance(angles, list):
-            angles = np.array(angles, dtype=np.float)
+        angles = np.array(angles, dtype=np.float)
         assert angles.shape == (3, )
         self._angles = angles
 
     def __repr__(self):
         return "Box(mins={}, maxs={}, angles={})".format(self.mins, self.maxs, self.angles)
+
+class boxArray(np.ndarray):
```

Comment:
Extremely nitpicky but PEP8 recommends CapCase for class names

https://www.python.org/dev/peps/pep-0008/#class-names